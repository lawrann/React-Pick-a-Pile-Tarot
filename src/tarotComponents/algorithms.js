function getZodiacSigns(year) {
  if (year < 4) return "Invalid year";
  switch ((year - 4) % 12) {
    case 0:
      return "Rat";

    case 1:
      return "Ox";

    case 2:
      return "Tiger";

    case 3:
      return "Rabbit";

    case 4:
      return "Dragon";

    case 5:
      return "Snake";

    case 6:
      return "Horse";

    case 7:
      return "Goat";

    case 8:
      return "Monkey";

    case 9:
      return "Rooster";

    case 10:
      return "Dog";

    case 11:
      return "Pig";
  }
}

function getHoroscopeSign(day, month) {
  var zodiacSigns = {
    capricorn: "Capricorn - Козерог",
    aquarius: "Aquarius - Водолей",
    pisces: "Pisces - Рыбы",
    aries: "Aries - Овен",
    taurus: "Taurus - Телец",
    gemini: "Gemini - Близнецы",
    cancer: "Cancer - Рак",
    leo: "Leo - Лев",
    virgo: "Virgo - Девы",
    libra: "Libra - Весы",
    scorpio: "Scorpio - Скорпион",
    sagittarius: "Sagittarius - Стрелец",
  };

  var cuspSigns = {
    mystery: "Cusp of Mystery - (Capricorn-Aquarius) / (Козерог-Водолей)", // Jan 16-23
    sensitivity: "Cusp of Sensitivity / (Aquarius-Pisces) / (Водолей-Рыбы)", // Feburary 15-21
    rebirth: "Cusp of Rebirth - (Pisces-Aries) / (Рыбы-Овен)", // March 17-23
    power: "Cusp of Power - (Aries-Taurus) / (Овен-Телец)", // April 16–22
    energy: "Cusp of Energy - (Taurus-Gemini) / (Телец-Близнецы)", // May 17–23
    magic: "Cusp of Magic - (Gemini-Cancer) / (Близнецы-Рак)", // June 17–23
    oscillation: "Cusp of Oscillation - (Cancer-Leo) / (Рак-Лев)", // July 19-25
    exposure: "Cusp of Exposure - (Leo-Virgo) / (Лев-Девы)", // Aug 19-25
    beauty: "Cusp of Beauty - (Virgo-Libra) / (Девы-Весы)", // Sep 19-25
    drama: "Cusp of Drama - (Libra-Scorpio) / (Весы-Скорпион)", // Oct 19-25
    revolution:
      "Cusp of Revolution - (Scorpio-Sagittarius) / (Скорпион-Стрелец)", // Nov 18-24
    prophecy: "Cusp of Prophecy - (Saggittarius-Capricorn) / (Стрелец-Козерог)", // Dec 18-24
  };
  if (month == 1 && day >= 16 && day <= 23) return cuspSigns.mystery;
  if (month == 2 && day >= 15 && day <= 21) return cuspSigns.sensitivity;
  if (month == 3 && day >= 17 && day <= 23) return cuspSigns.rebirth;
  if (month == 4 && day >= 16 && day <= 22) return cuspSigns.power;
  if (month == 5 && day >= 17 && day <= 23) return cuspSigns.energy;
  if (month == 6 && day >= 17 && day <= 23) return cuspSigns.magic;
  if (month == 7 && day >= 19 && day <= 25) return cuspSigns.oscillation;
  if (month == 8 && day >= 19 && day <= 25) return cuspSigns.exposure;
  if (month == 9 && day >= 19 && day <= 25) return cuspSigns.beauty;
  if (month == 10 && day >= 19 && day <= 25) return cuspSigns.drama;
  if (month == 11 && day >= 18 && day <= 24) return cuspSigns.revolution;
  if (month == 12 && day >= 18 && day <= 24) return cuspSigns.prophecy;

  if ((month == 1 && day <= 20) || (month == 12 && day >= 22)) {
    return zodiacSigns.capricorn;
  } else if ((month == 1 && day >= 21) || (month == 2 && day <= 18)) {
    return zodiacSigns.aquarius;
  } else if ((month == 2 && day >= 19) || (month == 3 && day <= 20)) {
    return zodiacSigns.pisces;
  } else if ((month == 3 && day >= 21) || (month == 4 && day <= 20)) {
    return zodiacSigns.aries;
  } else if ((month == 4 && day >= 21) || (month == 5 && day <= 20)) {
    return zodiacSigns.taurus;
  } else if ((month == 5 && day >= 21) || (month == 6 && day <= 20)) {
    return zodiacSigns.gemini;
  } else if ((month == 6 && day >= 22) || (month == 7 && day <= 22)) {
    return zodiacSigns.cancer;
  } else if ((month == 7 && day >= 23) || (month == 8 && day <= 23)) {
    return zodiacSigns.leo;
  } else if ((month == 8 && day >= 24) || (month == 9 && day <= 23)) {
    return zodiacSigns.virgo;
  } else if ((month == 9 && day >= 24) || (month == 10 && day <= 23)) {
    return zodiacSigns.libra;
  } else if ((month == 10 && day >= 24) || (month == 11 && day <= 22)) {
    return zodiacSigns.scorpio;
  } else if ((month == 11 && day >= 23) || (month == 12 && day <= 21)) {
    return zodiacSigns.sagittarius;
  }
  return "Invalid birthday";
}

function getNameNumerology(name) {
  var x = {
    A: 1,
    J: 1,
    S: 1,
    B: 2,
    K: 2,
    T: 2,
    C: 3,
    L: 3,
    U: 3,
    D: 4,
    M: 4,
    V: 4,
    E: 5,
    N: 5,
    W: 5,
    F: 6,
    O: 6,
    X: 6,
    G: 7,
    P: 7,
    Y: 7,
    H: 8,
    Q: 8,
    Z: 8,
    I: 9,
    R: 9,
  };

  var nameVal = 0;
  for (var i = 0; i < name.length; i++) {
    var curChar = name.charAt(i).toUpperCase();
    if (curChar === " " || curChar === ",") {
      continue;
    }
    var curValue = x[curChar];
    nameVal = nameVal + curValue;
  }
  var singleDigitScore = nameVal;
  while (singleDigitScore >= 10) {
    var total = 0;
    var str = "" + singleDigitScore;
    for (var j = 0; j < str.length; j++) {
      total = total + parseInt(str.charAt(j));
    }
    singleDigitScore = total;
  }
  return [nameVal, singleDigitScore];
}

function shuffle(array) {
  var currentIndex = array.length,
    temporaryValue,
    randomIndex;

  // While there remain elements to shuffle...
  while (0 !== currentIndex) {
    // Pick a remaining element...
    randomIndex = Math.floor(Math.random() * currentIndex);
    currentIndex -= 1;

    // And swap it with the current element.
    temporaryValue = array[currentIndex];
    array[currentIndex] = array[randomIndex];
    array[randomIndex] = temporaryValue;
  }

  return array;
}

export { getHoroscopeSign, getZodiacSigns, getNameNumerology, shuffle };
