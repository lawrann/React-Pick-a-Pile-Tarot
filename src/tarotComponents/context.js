import React from "react";

const context = React.createContext({});
const ContextProvider = context.Provider;
const ContextConsumer = context.Consumer;

export { ContextProvider, ContextConsumer };
