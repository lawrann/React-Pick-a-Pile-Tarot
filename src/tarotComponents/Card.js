import React from "react";
import { ContextConsumer } from "./context";

const Card = (prop) => {
  return (
    <ContextConsumer>
      {({ actions, state }) => {
        function cardBack() {
          var images = [
            require("./cards/" +
              state.tarot[state.piles[prop.id]["cardId"]].img),
            require("./cards/cardback.jpg"),
          ];
          if (state.piles[prop.id]["display"]) {
            return images[0];
          } else {
            return images[1];
          }
        }
        return (
          <React.Fragment>
            <img
              // src={require("./cards/cardback.jpg")}
              src={cardBack()}
              width="200px"
              height="320px"
              onClick={() => actions.toggleDisplay(prop.id)}
              id={"cardImg" + prop.id}
            />
          </React.Fragment>
        );
      }}
    </ContextConsumer>
  );
};

export default Card;
