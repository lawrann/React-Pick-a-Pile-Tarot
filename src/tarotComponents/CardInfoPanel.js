import React from "react";
import { ContextConsumer } from "./context";

const CardInfoPanel = () => {
  return (
    <ContextConsumer>
      {({ actions, state }) => {
        return (
          <React.Fragment>
            <h1>Cards selected:</h1>
            <table>{actions.getCardInfoDisplay()}</table>
          </React.Fragment>
        );
      }}
    </ContextConsumer>
  );
};

export default CardInfoPanel;
