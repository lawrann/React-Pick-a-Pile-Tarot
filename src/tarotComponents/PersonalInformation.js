import React from "react";
import { ContextConsumer } from "./context";
import { Row, Col, Container, Form } from "react-bootstrap";
import "bootstrap/dist/css/bootstrap.min.css";

const PersonalInformation = () => {
  return (
    <ContextConsumer>
      {({ actions }) => {
        return (
          <React.Fragment>
            <Container>
              <Form>
                <Row>
                  <Col>
                    <Form.Group>
                      <Form.Label>Full Name:</Form.Label>
                      <Form.Control
                        onChange={actions.getName}
                        id="nameBox"
                        type="text"
                        placeholder="Enter your Name"
                      />
                      <Form.Text className="text-muted">
                        You name is used to calculate your Name's Numerlogy
                      </Form.Text>
                    </Form.Group>
                  </Col>
                  <Col>
                    <Form.Group>
                      <Form.Label>Date of Birth:</Form.Label>
                      <Form.Control
                        type="date"
                        id="birthdayBox"
                        onChange={actions.getBirthday}
                        placeholder="Enter your Date of Birth"
                      />
                      <Form.Text className="text-muted">
                        You DOB is used to calculate your Horoscope and Zodiac
                        signs
                      </Form.Text>
                    </Form.Group>
                  </Col>
                </Row>
                <Row>
                  <Col>Name Numerology: </Col>
                  {actions.displayNumber()}
                </Row>
                <Row>
                  <Col>Horoscope: </Col>
                  {actions.displayHoroscope()}
                </Row>
                <Row>
                  <Col>Chinese Zodiac: </Col>
                  {actions.displayZodiac()}
                </Row>
              </Form>
            </Container>
            {/* <div>
              Full Name:{" "}
              <input
                type="text"
                placeholder="Full Name"
                id="nameBox"
                onChange={actions.getName}
              ></input>
            </div> */}
            {/* <div>
              Birthday:{" "}
              <input
                type="date"
                id="birthdayBox"
                onChange={actions.getBirthday}
              ></input>
            </div> */}
            {/* <div>Name Numerology: {actions.displayNumber()}</div>
            <div>Horoscope: {actions.displayHoroscope()}</div>
            <div>Chinese Zodiac: {actions.displayZodiac()}</div> */}
          </React.Fragment>
        );
      }}
    </ContextConsumer>
  );
};

export default PersonalInformation;
