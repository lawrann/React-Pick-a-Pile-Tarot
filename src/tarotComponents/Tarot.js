import React from "react";
import { ContextProvider } from "./context";
import TarotData from "./tarot-images.json";
import * as algo from "./algorithms";
import "./tarotStyles.css";
import { Row, Col, Container, Form } from "react-bootstrap";
import "bootstrap/dist/css/bootstrap.min.css";

class Tarot extends React.Component {
  state = {
    tarot: TarotData.cards,
    cardPicked: [],
    name: "",
    birthday: "",
    horoscope: "",
    zodiac: "",
    nameNumerology: "",
    singleNumber: "",
    generatedPiles: false,
    number_piles: 35,
    piles: [
      { pile: 1, cardId: -1, display: false },
      { pile: 2, cardId: -1, display: false },
      { pile: 3, cardId: -1, display: false },
      { pile: 4, cardId: -1, display: false },
    ],
    cardsSelected: [],
  };

  getCardSelectedDisplay = () => {
    let uis = [];
    var num = this.state.number_piles;
    let card_true = [];
    for (var i = 0; i < num; i++) {
      if (this.state.piles[i]["display"] === true) {
        card_true.push(this.state.piles[i]["cardId"]);
      }
    }
    for (var i = 0; i < card_true.length; i++) {
      if (this.state.cardsSelected.includes(card_true[i])) {
        continue;
      } else {
        this.state.cardsSelected.push(card_true[i]);
      }
    }
    for (var i = 0; i < this.state.cardsSelected.length; i++) {
      if (card_true.includes(this.state.cardsSelected[i])) {
        continue;
      } else {
        this.state.cardsSelected.splice(i, 1);
      }
    }
    var count = 0;
    for (const cardid of this.state.cardsSelected) {
      uis.push(
        <img
          data-testid={"card-selected-id-" + count}
          className="cards-selected"
          src={require("./cards/" + this.state.tarot[cardid].img)}
          width="150px"
          height="240px"
          key={"cardid-" + count}
        />
      );
      count++;
    }
    return uis;
  };

  getCardInfoDisplay = () => {
    let uis = [];
    var num = this.state.number_piles;
    let card_true = [];
    for (var i = 0; i < num; i++) {
      if (this.state.piles[i]["display"] === true) {
        card_true.push(this.state.piles[i]["cardId"]);
      }
    }
    for (var i = 0; i < card_true.length; i++) {
      if (this.state.cardsSelected.includes(card_true[i])) {
        continue;
      } else {
        this.state.cardsSelected.push(card_true[i]);
      }
    }
    for (var i = 0; i < this.state.cardsSelected.length; i++) {
      if (card_true.includes(this.state.cardsSelected[i])) {
        continue;
      } else {
        this.state.cardsSelected.splice(i, 1);
      }
    }
    for (const cardid of this.state.cardsSelected) {
      uis.push(this.getCardData(cardid));
    }
    console.log("cardsSelected " + this.state.cardsSelected);

    return uis;
  };

  toggleDisplay = (pileNum) => {
    console.log("Select card index: " + pileNum);
    console.log("Tarot card number: " + this.state.piles[pileNum]["cardId"]);
    console.log(
      "Tarot card: " +
        this.state.tarot[this.state.piles[pileNum]["cardId"]].name
    );
    this.state.piles[pileNum]["display"] = !this.state.piles[pileNum][
      "display"
    ];

    this.setState((state) => ({
      piles: this.state.piles,
    }));
  };

  displayNumber = () => {
    return (
      <Col data-testid="namenumerlogy">
        {this.state.nameNumerlogy} / {this.state.singleNumber}
      </Col>
    );
  };

  displayHoroscope = () => {
    return <Col data-testid="horoscope">{this.state.horoscope}</Col>;
  };

  displayZodiac = () => {
    return <Col data-testid="zodiac">{this.state.zodiac}</Col>;
  };

  getName = () => {
    var n = document.getElementById("nameBox").value;
    var numerlogy = algo.getNameNumerology(n);
    this.setState(() => ({
      name: n,
      nameNumerlogy: numerlogy[0],
      singleNumber: numerlogy[1],
    }));
    console.log("name: " + this.state.name);
  };

  getBirthday = () => {
    var b = document.getElementById("birthdayBox").value;
    var month = b.split("-")[1];
    var day = b.split("-")[2];
    var year = b.split("-")[0];
    this.setState(() => ({
      birthday: b,
      horoscope: algo.getHoroscopeSign(day, month),
      zodiac: algo.getZodiacSigns(year),
    }));
  };

  getAnswers = (answer, question, key) => {
    let uis = [];
    if (answer === "" || answer === undefined) {
      return;
    } else if (typeof answer === "object") {
      uis.push(<b key={"1-" + key + "-" + question}>{question}</b>);
      uis.push(<br key={"2-" + key + "-" + question} />);
      var count = 0;
      for (const q of answer) {
        count++;
        uis.push(q);
        uis.push(<br key={"3-" + key + "-" + question + "-" + count} />);
      }
    } else {
      uis.push(<b key={"4-" + key + "-" + question}>{question}</b>);
      uis.push(<br key={"5-" + key + "-" + question} />);
      uis.push(answer);
      uis.push(<br key={"6-" + key + "-" + question} />);
    }
    uis.push(<br key={"7-" + key + "-" + question} />);
    return uis;
  };

  // getCardId = (pileNum) => {
  //   const clonePile = [...this.state.piles];
  //   clonePile[pileNum]["display"] = !clonePile[pileNum]["display"];
  //   this.setState((prevState) => ({
  //     generatedPiles: true,
  //     piles: [...prevState.piles],
  //   }));
  //   return this.state.piles[pileNum]["cardId"];
  // };

  getCardData = (cardNumber) => {
    return (
      <React.Fragment key={cardNumber}>
        <tbody data-testid={"card-data-tbody-id-" + cardNumber}>
          <tr>
            <td>
              <h3>
                {this.state.tarot[cardNumber].name} (
                {this.state.tarot[cardNumber].arcana})
              </h3>
              <img
                data-testid={"card-data-img-id-" + cardNumber}
                // onClick={() => this.getCardId(cardNumber)}
                src={require("./cards/" + this.state.tarot[cardNumber].img)}
                width="300px"
                height="480px"
              />
            </td>
            <td>
              <p>
                {this.getAnswers(
                  this.state.tarot[cardNumber].fortune_telling,
                  "Fortune Telling:",
                  cardNumber + "-" + 1
                )}
                {this.getAnswers(
                  this.state.tarot[cardNumber]["Questions to Ask"],
                  "Questions to Ask:",
                  cardNumber + "-" + 2
                )}
                {this.getAnswers(
                  this.state.tarot[cardNumber].keywords,
                  "Keywords:",
                  cardNumber + "-" + 3
                )}
              </p>
            </td>
            <td>
              <p>
                {this.getAnswers(
                  this.state.tarot[cardNumber].meanings.light,
                  "Light Meaning:",
                  cardNumber + "-" + 4
                )}
                {this.getAnswers(
                  this.state.tarot[cardNumber].meanings.shadow,
                  "Shadow Meaning:",
                  cardNumber + "-" + 5
                )}
              </p>
            </td>
            <td>
              <p>
                {this.getAnswers(
                  this.state.tarot[cardNumber].number,
                  "Number",
                  cardNumber + "-" + 6
                )}
                {this.getAnswers(
                  this.state.tarot[cardNumber].suit,
                  "Suit",
                  cardNumber + "-" + 7
                )}
                {this.getAnswers(
                  this.state.tarot[cardNumber]["Mythical/Spiritual"],
                  "Mythical/Spiritual:",
                  cardNumber + "-" + 8
                )}
                {this.getAnswers(
                  this.state.tarot[cardNumber].Archetype,
                  "Archetype:",
                  cardNumber + "-" + 9
                )}
                {this.getAnswers(
                  this.state.tarot[cardNumber].Elemental,
                  "Elemental:",
                  cardNumber + "-" + 10
                )}
                {this.getAnswers(
                  this.state.tarot[cardNumber].Affirmation,
                  "Affirmation",
                  cardNumber + "-" + 11
                )}
              </p>
            </td>
          </tr>
        </tbody>
      </React.Fragment>
    );
  };

  generateNewCards = () => {
    // Shuffle deck a random number of times or based on the name numerlogy of the person
    var deck;
    if (this.state.singleNumber === "") {
      let numTimes = Math.floor(Math.random() * 10);
      deck = this.state.tarot;
      var count = 0;
      while (count !== numTimes) {
        deck = algo.shuffle(deck);
        count++;
      }
    } else {
      let numTimes = this.state.singleNumber;
      deck = this.state.tarot;
      var count = 0;
      while (count !== numTimes) {
        deck = algo.shuffle(deck);
        count++;
      }
    }
    let cardArr = [this.state.number_piles];
    var pass_flag = true;

    // Ensuring each card is unique
    do {
      cardArr = [this.state.number_piles];
      pass_flag = true;
      for (var i = 0; i < this.state.number_piles; i++) {
        var newCard = Math.floor(Math.random() * 78);
        if (!cardArr.includes(newCard)) {
          cardArr.push(newCard);
          continue;
        } else {
          pass_flag = false;
          break;
        }
      }
    } while (!pass_flag);
    console.log("final cardArr " + cardArr);

    // add pulls to state piles
    var curPileLen = Object.keys(this.state.piles).length;
    for (var i = 0; i < this.state.number_piles; i++) {
      if (i < curPileLen) {
        this.state.piles[i]["cardId"] = cardArr[i + 1];
        this.state.piles[i]["display"] = false;
      } else {
        this.state.piles.push({
          pile: i,
          cardId: cardArr[i + 1],
          display: false,
        });
      }
    }

    this.setState(() => ({
      generatedPiles: true,
      cardsSelected: [],
      tarot: deck,
    }));
  };

  render() {
    return (
      <React.Fragment>
        <h1 className="primary">Pick a Pile Tarot Reading</h1>
        <ContextProvider
          value={{
            state: this.state,
            tarot: this.tarot,
            cardPicked: this.cardPicked,
            name: this.name,
            birthday: this.birthday,
            horoscope: this.horoscope,
            zodiac: this.zodiac,
            piles: this.piles,
            generatedPiles: this.generatedPiles,
            number_piles: this.number_piles,
            actions: {
              getCardData: this.getCardData,
              getName: this.getName,
              getBirthday: this.getBirthday,
              displayNumber: this.displayNumber,
              displayHoroscope: this.displayHoroscope,
              displayZodiac: this.displayZodiac,
              generateNewCards: this.generateNewCards,
              getCardId: this.getCardId,
              toggleDisplay: this.toggleDisplay,
              getAnswers: this.getAnswers,
              getCardInfoDisplay: this.getCardInfoDisplay,
              getCardSelectedDisplay: this.getCardSelectedDisplay,
            },
          }}
        >
          {this.props.children}
        </ContextProvider>
      </React.Fragment>
    );
  }
}

export default Tarot;
