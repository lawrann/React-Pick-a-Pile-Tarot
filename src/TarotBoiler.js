import React from "react";
import Tarot from "./tarotComponents/Tarot";
import PersonalInformation from "./tarotComponents/PersonalInformation";
import CardsGenerator from "./tarotComponents/CardsGenerator";

const TarotBoiler = () => {
  return (
    <div className="tarotBoiler">
      <Tarot>
        <PersonalInformation />
        <CardsGenerator />
      </Tarot>
    </div>
  );
};

export default TarotBoiler;
