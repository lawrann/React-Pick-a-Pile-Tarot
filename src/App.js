import React from "react";
import TarotBoiler from "./TarotBoiler";

function App() {
  return (
    <div className="App">
      <TarotBoiler />
    </div>
  );
}

export default App;
