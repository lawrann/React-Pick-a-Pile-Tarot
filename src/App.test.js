import React from "react";
import App from "./App";
import TarotBoiler from "./TarotBoiler";
import Tarot from "./tarotComponents/Tarot";
import { ContextProvider, ContextConsumer } from "./tarotComponents/context";
import Card from "./tarotComponents/Card";
import PersonalInformation from "./tarotComponents/PersonalInformation";
import renderer from "react-test-renderer";
import {
  getNameNumerology,
  getZodiacSigns,
  getHoroscopeSign,
  shuffle,
} from "./tarotComponents/algorithms";

// how to test for onchange function?

describe("tarot functions", () => {
  test("getCardData", () => {
    const cardData = renderer.create(
      <Tarot>
        <ContextConsumer>
          {({ actions }) => {
            {
              actions.getCardData(1);
            }
          }}
        </ContextConsumer>
      </Tarot>
    );
    let tree = cardData.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("test context provider", () => {
  it("sets generatedpiles status to false", () => {
    const TestComponent = () => {
      return (
        <ContextConsumer>
          {({ actions, state }) => {
            <div data-testid="value">
              {state.generatedPiles.toString()}
              <button onClick={actions.generateNewCards}>
                generate new cards
              </button>
            </div>;
          }}
        </ContextConsumer>
      );
    };

    const tarotBoilerComponent = renderer.create(
      <Tarot>
        <TestComponent />
      </Tarot>
    );

    let tree = tarotBoilerComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("test algorithms", () => {
  test("name numerology", () => {
    expect(getNameNumerology("A")).toStrictEqual([1, 1]);
    expect(getNameNumerology("B")).toStrictEqual([2, 2]);
    expect(getNameNumerology("R")).toStrictEqual([9, 9]);
    expect(getNameNumerology("AB")).toStrictEqual([3, 3]);
  });

  test("zodiac signs", () => {
    expect(getZodiacSigns("1995")).toStrictEqual("Pig");
    expect(getZodiacSigns("13994")).toStrictEqual("Dog");
    expect(getZodiacSigns("-1")).toStrictEqual("Invalid year");
  });

  test("horoscope signs", () => {
    expect(getHoroscopeSign("26", "6")).toStrictEqual("Cancer - Рак");
    expect(getHoroscopeSign("25", "8")).toStrictEqual(
      "Cusp of Exposure - (Leo-Virgo) / (Лев-Девы)"
    );
    expect(getHoroscopeSign("-1", "-1")).toStrictEqual("Invalid birthday");
  });

  test("shuffle", () => {
    const v_arr = [1, 2, 3, 4, 5, 6];
    expect(shuffle(v_arr)).toHaveLength(v_arr.length);
  });
});

describe("snapshot testing", () => {
  test("TarotBoiler render", () => {
    const tarotBoilerComponent = renderer.create(<TarotBoiler />);
    let tree = tarotBoilerComponent.toJSON(); // toJson is to create the snapshot
    expect(tree).toMatchSnapshot();
  });

  test("Tarot render", () => {
    const tarotComponent = renderer.create(<Tarot />);
    let tree = tarotComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

//   test("Card render", () => {
//     const cardComponent = renderer.create(<Card />);
//     let tree = cardComponent.toJSON();
//     expect(tree).toMatchSnapshot();
//   });

//   test("PersonalInformation render", () => {
//     const component = renderer.create(<PersonalInformation />);
//   });

// // manually trigger the callback
// tree.props.onMouseEnter();
// // re-rendering
// tree = component.toJSON();
// expect(tree).toMatchSnapshot();

// // manually trigger the callback
// tree.props.onMouseLeave();
// // re-rendering
// tree = component.toJSON();
// expect(tree).toMatchSnapshot();

// test("", () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

// Snapshot test case renders a UI component, takes a snapshot,
// then compares it to a reference snapshot file stored alongside the test.
// The test will fail if the two snapshots do not match: either the change is unexpected,
// or the reference snapshot needs to be updated to the new version of the UI component.
// test("the pizza data is correct", () => {
//   expect(pizzas).toMatchSnapshot();
// });

// react-testing
// it("renders correctly", () => {
//   const { getByTestId, queryByPlaceholderName } = render(<TarotBoiler />);
//   expect(getByTestId("tarot")).toBeTruthy();
// });
