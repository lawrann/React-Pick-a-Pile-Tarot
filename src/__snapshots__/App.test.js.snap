// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing Card render 1`] = `
<h1
  className="primary"
>
  Pick a Pile Tarot Reading
</h1>
`;

exports[`snapshot testing Tarot render 1`] = `
<h1
  className="primary"
>
  Pick a Pile Tarot Reading
</h1>
`;

exports[`snapshot testing TarotBoiler render 1`] = `
<div
  className="tarotBoiler"
>
  <h1
    className="primary"
  >
    Pick a Pile Tarot Reading
  </h1>
  <div
    className="container"
  >
    <form
      className=""
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Full Name:
            </label>
            <input
              className="form-control"
              id="nameBox"
              onChange={[Function]}
              placeholder="Enter your Name"
              type="text"
            />
            <small
              className="text-muted form-text"
            >
              You name is used to calculate your Name's Numerlogy
            </small>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Date of Birth:
            </label>
            <input
              className="form-control"
              id="birthdayBox"
              onChange={[Function]}
              placeholder="Enter your Date of Birth"
              type="date"
            />
            <small
              className="text-muted form-text"
            >
              You DOB is used to calculate your Horoscope and Zodiac signs
            </small>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          Name Numerology: 
        </div>
        <div
          className="col"
        >
          Numerlogy: 
           / 
          
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          Horoscope: 
        </div>
        <div
          className="col"
        >
          
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          Chinese Zodiac: 
        </div>
        <div
          className="col"
        >
          
        </div>
      </div>
    </form>
  </div>
  <button
    className="btn btn-primary btn-sm m-2 btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Generate Cards
  </button>
  <div
    className="fade alert alert-primary show"
    role="alert"
  >
    Click on generate cards first!
  </div>
</div>
`;

exports[`snapshot testing TarotBoiler test 1`] = `
<div
  className="tarotBoiler"
>
  <h1
    className="primary"
  >
    Pick a Pile Tarot Reading
  </h1>
  <div
    className="container"
  >
    <form
      className=""
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Full Name:
            </label>
            <input
              className="form-control"
              id="nameBox"
              onChange={[Function]}
              placeholder="Enter your Name"
              type="text"
            />
            <small
              className="text-muted form-text"
            >
              You name is used to calculate your Name's Numerlogy
            </small>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Date of Birth:
            </label>
            <input
              className="form-control"
              id="birthdayBox"
              onChange={[Function]}
              placeholder="Enter your Date of Birth"
              type="date"
            />
            <small
              className="text-muted form-text"
            >
              You DOB is used to calculate your Horoscope and Zodiac signs
            </small>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          Name Numerology: 
        </div>
        <div
          className="col"
        >
          Numerlogy: 
           / 
          
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          Horoscope: 
        </div>
        <div
          className="col"
        >
          
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          Chinese Zodiac: 
        </div>
        <div
          className="col"
        >
          
        </div>
      </div>
    </form>
  </div>
  <button
    className="btn btn-primary btn-sm m-2 btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Generate Cards
  </button>
  <div
    className="fade alert alert-primary show"
    role="alert"
  >
    Click on generate cards first!
  </div>
</div>
`;

exports[`tarot functions getCardData 1`] = `
<h1
  className="primary"
>
  Pick a Pile Tarot Reading
</h1>
`;

exports[`test context provider sets generatedpiles status to false 1`] = `
<h1
  className="primary"
>
  Pick a Pile Tarot Reading
</h1>
`;
